.azure-pipelines/sync-tools-folder.yml

```diff
@@ -1,56 +1,68 @@
-resources:
-  repositories:
-  - repository: self
-    type: git
-    ref: main
+parameters:
+- name: ToBranches
+  displayName: To Branches
+  type: object
+  default: ["generation"]
+- name: SyncPaths
+  displayName: Paths to sync
+  type: object
+  default:
+  - .ci-config.json
+  - tools
+  - .azure-pipelines
+  - src/lib
+  - src/shared
+  - build.proj
+  - src/Az.props
+  - src/Az.Shared.props
+  - src/Az.Test.props
+  - src/Az.autorest.props
+  - src/Az.Post.props
+  - src/psm1.props
+  - NuGet.Config
 
 trigger:
   branches:
     include:
       - main
   paths:
     include:
-      - .azure-pipelines
-      - .devcontainer
-      - tools
-      - src/lib
-      - .ci-config.json
-      - src/shared
-      - build.proj
-      - src/Az.props
-      - src/Az.Shared.props
-      - src/Az.Test.props
-      - src/Az.autorest.props
-      - src/Az.Post.props
-      - src/psm1.props
-      - NuGet.Config
+    - .ci-config.json
+    - tools
+    - .azure-pipelines
+    - src/lib
+    - src/shared
+    - build.proj
+    - src/Az.props
+    - src/Az.Shared.props
+    - src/Az.Test.props
+    - src/Az.autorest.props
+    - src/Az.Post.props
+    - src/psm1.props
+    - NuGet.Config
       
-variables:
-  GenerationBranchName: generation
-
 jobs:
-- job: Sync
-  displayName: Sync task
-  condition: succeeded()
-  strategy:
-    matrix:
-      Generation:
-        BranchName: ${{ variables.GenerationBranchName }}
+- ${{ each ToBranch in parameters.ToBranches }}:
+    - job: Sync_${{ replace(ToBranch, '/', '_') }}
+      displayName: "Sync To ${{ ToBranch }}"
+      pool: pool-ubuntu-2004
 
-  steps:
-  - template: util/get-github-pat-steps.yml
-  - task: PowerShell@2
-    displayName: Sync branch
-    inputs:
-      targetType: inline
-      script: >-
-        ./tools/SyncFromMainBranch.ps1 -BranchName $(BranchName) -GithubToken $(GithubToken)
-      pwsh: true
+      steps:
+      - template: util/get-github-pat-steps.yml
```

Comment:
this yml has github pat involved, why this was not moved to ADO?

Reply:
There are a few internal pipelines depending on `get-github-pat-steps.yml`. Last time we only remove the dependency from public pipelines. This token is used for several targets: checkout, creating PRs, getting PR files, ...

Reply:
I don't get it, any other yml under this folder also depend on github pat?